packages/mui-material/src/Modal/ModalManager.test.ts

```diff
@@ -386,35 +388,59 @@ describe('ModalManager', () => {
     });
 
     it('should add aria-hidden to previous modals', () => {
-      const modal2 = document.createElement('div');
-      const modal3 = document.createElement('div');
-
-      container2.appendChild(modal2);
-      container2.appendChild(modal3);
-
-      modalManager.add({ ...getDummyModal(), modalRef: modal2 }, container2);
+      const modal2: Modal = { mount: container2, modalRef: document.createElement('div') };
+      modal2.modalRef.id = 'modal2Ref';
+      const modal3: Modal = { mount: container2, modalRef: document.createElement('div') };
+      modal3.modalRef.id = 'modal3Ref';
+
+      container2.appendChild(modal2.modalRef);
+      container2.appendChild(modal3.modalRef);
+
+      // ideally mount must be a child of container, and modalRef must be a child of mount.
+      // usually mount is worked out out by Portal:
+      // * if disablePortal is true, mount is modalRef or null (even though types don't allow that).
+      //   it could be the container2 because it is technically the mount (although it doesn't happen in real scenarios).
+      // * if disablePortal is false, mount is the container if set, or document.body.
+
+      // We directly mounted the modalRef to the container2, so
+      // mount is modalRef or the container too (or null, but typing doesn't allow it).
```

Comment:
I removed the comment it was originally a note to myself on what should I need to handle inside `ModalManager`.

as it says it doesn't happen in real scenarios, but what we usually expect from the modal object is that the DOM order resolves to something like this. `container`->`mountNodeRef`->`modalRef`. From reading the codebase (before my fix) it could happen that `mountNodeRef` resolves to either something unique, `null`, `modalRef` or the `container` itself. So all cases were handled.

After that I added new tests instead of changing this but some changes remained to it, so I reverted them too.